nixos/modules/services/web-apps/discourse.nix

```diff
@@ -0,0 +1,120 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.discourse;
+
+  bundle = "${cfg.package}/share/discourse/bin/bundle";
+
+in
+
+{
+  options = {
+    services.discourse = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable the Discourse service.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.discourse;
+        defaultText = "pkgs.discourse";
+        description = ''
+          Which Discourse package to use.
+        '';
+        example = "pkgs.discourse.override { ruby = pkgs.ruby_2_5; }";
+      };
+
+      port = mkOption {
+        type = types.int;
+        default = 3000;
+        description = "Port on which Discourse is ran.";
+      };
+
+      stateDir = mkOption {
+        type = types.str;
+        default = "/var/lib/discourse";
+        description = "The state directory, logs and plugins are stored here.";
+      };
+
+      database = {
+
+        host = mkOption {
+          type = types.str;
+          default = (if cfg.database.socket != null then "localhost" else "127.0.0.1");
+          description = "Database host address.";
+        };
+
+        port = mkOption {
+          type = types.int;
+          default = 5432;
+          description = "Database host port.";
+        };
+
+        name = mkOption {
+          type = types.str;
+          default = "discourse";
+          description = "Database name.";
+        };
+
+        user = mkOption {
+          type = types.str;
+          default = "discourse";
+          description = "Database user.";
+        };
+
+        password = mkOption {
+          type = types.str;
+          default = "";
+          description = ''
+            The password corresponding to <option>database.user</option>.
+            Warning: this is stored in cleartext in the Nix store!
+            Use <option>database.passwordFile</option> instead.
+          '';
+        };
+
+        passwordFile = mkOption {
+          type = types.nullOr types.path;
```

Comment:
Should be `nullOr str` to avoid the file ending up in the nix store if users set `services.discourse.database.passwordFile = /run/keys/discourse-dbpassword`